recipes/c4core/all/conanfile.py

```diff
@@ -0,0 +1,84 @@
+    description = "a library of low-level C++ utilities, written with low-latency projects in mind."
+    topics = ("utilities", "low-latency", )
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/mutouyun/cpp-ipc"
+    license = "MIT",
+    settings = "os", "arch", "compiler", "build_type"
+    exports_sources = ["CMakeLists.txt"]
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    generators = "cmake", "cmake_find_package_multi"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "11")
+
+        ## clang/apple-clang with libc++ is not supported. It is already fixed at 2022-01-03.
+        if tools.Version(self.version) <= "0.1.8":
+            if (self.settings.compiler == "clang" and self.settings.compiler.libcxx == "libc++"):
+                raise ConanInvalidConfiguration(self, "{}/{} doesn't support clang with libc++".format(self.name, self.version))
+            if (self.settings.compiler == "apple-clang" and self.settings.compiler.libcxx == "libc++"):
+                raise ConanInvalidConfiguration(self, "{}/{} doesn't support apple-clang with libc++".format(self.name, self.version))
+
+    def requirements(self):
+        self.requires("fast_float/3.4.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+            destination=self._source_subfolder, strip_root=True)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"), "c4/ext/fast_float_all.h", "")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "src", "c4", "ext", "fast_float.hpp"),
+            '#include "c4/ext/fast_float_all.h"',
+            '#include "fast_float/fast_float.h"')
+
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="LICENSE*", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.remove_files_by_mask(os.path.join(self.package_folder, "include"), "*.natvis")
+
+    def package_info(self):
+        self.cpp_info.libs = ["c4core"]
```

Comment:
```suggestion
        self.cpp_info.set_property("cmake_file_name", "c4core")
        self.cpp_info.set_property("cmake_target_name", "c4core::c4core")
        self.cpp_info.libs = ["c4core"]
```

enforce CMake names, otherwise it's hard for me to track in the 1000+ recipes if CMake names are properly models.